### ISSUE TITLE
external_links gets removed from config, it doesn't work

### ISSUE THREAD

**Issue by fbrbovic** [rancher/rancher#4095]
Rancher Version:
v0.63.0
Docker Version:
Version:      1.10.1
OS: 
RancherOS
Steps to Reproduce: 
Use following : 
```
web-data:
  labels:
    io.rancher.container.start_once: 'true'
  image: busybox
  volumes:
    - /data
web-conf:
  labels:
    io.rancher.container.hostname_override: container_name
  image: ubuntu
  volumes_from:
    - web-data
  external_links:
    - vault/vault:vault
web:
  ports:
    - 8081:80/tcp
  labels:
    io.rancher.sidekicks: web-data,web-conf
  image: ubuntu
  volumes_from:
    - web-data
```
Results: 
```
web-data:
  labels:
    io.rancher.container.start_once: 'true'
  image: busybox
  volumes:
    - /data
web-conf:
  labels:
    io.rancher.container.hostname_override: container_name
  image: ubuntu
  volumes_from:
    - web-data
web:
  ports:
    - 8081:80/tcp
  labels:
    io.rancher.sidekicks: web-data,web-conf
  image: ubuntu
  volumes_from:
    - web-data
```
Expected:

I set the above docker-compose.yml via gui when creating a new stack , but when service is started external link is never linked and it get remvoed from the config.  I do have that external link live and functioning

For some reason external_links are not set and disappear from the config

**Comment by deniseschannon** [comment]
In your setup, external links are set at the primary service (in your case `web` is considered the primary service as it has the `io.rancher.sidekicks` label), which is not supported.

In our next release (v.1.0.0.-rc2+), you could choose to link at the primary service and sidekicks will be able to resolve using the same link. In general, all services can resolve to any service in Rancher at `servicename.stackname.rancher.internal` without the need for links/externallinks.

**Comment by fbrbovic** [comment]
Actually in my example external_links was setup at the sidekick level at web-conf service. 

Are you saying that at any level as long as there are sidekick services its not possible to use external_links?
Also why is rancher not complaining about it instead it launches the stack but removes any 'external_links' flag from the stack config. I also checked the docker logs for rancher and no errors being reported. 

This is all strange since I also had a stack which was setup prior to v0.63.0  (v0.56.0 I believe)  which had both external_links and sidekicks and it worked and still works.  But for any new stacks it remvoes the 'external_links' flag

**Comment by deniseschannon** [comment]
> Actually in my example external_links was setup at the sidekick level at web-conf service.

This is exactly what is NOT supported. Sorry if I wasn't clear. We have only ever supported links/external_links on the primary service. 

I just tested in v0.56.1 and when I place the above compose file (where external links are on the sidekick), this does not create the external-link in rancher UI, so I don't believe this was ever working as you expected it to. In the UI, it doesn't allow for service links to be created for sidekick services. 

With our v1.0.0-rcX releases, we've also enabled cross stack linking in the DNS, so no need to explicitly set an external_link to get DNS resolution. But if you set a link at the primary service level, the sidekick would also be able to resolve using the same link. This would be used if you needed to put in a servicealias and couldn't reference the service name for some reason.
